lib/shopify_api/resources/smart_collection.rb

```diff
@@ -3,13 +3,17 @@ class SmartCollection < Base
     include Events
     include Metafields
 
-    def products
-      Product.find(:all, :params => {:collection_id => self.id})
+    def products(options={})
+      if options.has_key?(:only_sorted)
+        get(:products, options).map { |p| Product.new(p) }
```

Comment:
There must be a more effective way to do this

Reply:
`ShopifyAPI::Product.find(:all, from: "/admin/smart_collections/#{id}/products.json)` I think

Reply:
but, I think this will cause issues if there are more than 50 products. We may need to parameterize `limit`, `page`, and/or `since_id`

Reply:
`Product.find()` won't use the products controller if you specify the `from:` parameter

Reply:
So Jamie, this does go to the heart of the question I raised in the PR description -- do people who use the existing products method in this Gem expect to be able to add all of the same filters that can be added to the Product API when a collection_id is specified? Limit will work but page and offset will not as this endpoint will use a different pagination method. the only way I can think to work around that is to create a different name for this new endpoint, which isn't products. Or to make a really hacky version of our new endpoint that redirects to the profuct controller if we receive params we don't support.  

I suppose we don't ever have to switch the Gem to use the new endpoint for just a straight call to products, but then if they want to receive all products sorted accurately according to the storefront order which has manualyl sorted products first they'll need to use `only_sorted:all` in the params to force it. I suppose its an ok workaround, but not ideal.

@DrewMartin @ignacio-chiazzo thoughts?

Reply:
> `Product.find()` won't use the products controller if you specify the `from:` parameter

Right, I should have read your response more closely, thanks!

Reply:
> do people who use the existing products method in this Gem expect to be able to add all of the same filters that can be added to the Product API when a collection_id is specified? 

I think it's definitely a reasonable expectation, but either way we'll need to provide some interface in the gem to paginate in the case where there might be > 250 products in a collection.

Hows does pagination work in the new endpoint?

Reply:
I'll send you the design doc so you can see. We're also creating a migration page to explain it because its a new concept for the REST API.